cmd/hack/hack.go

```diff
@@ -1666,6 +1681,243 @@ func GenerateTxLookups1(chaindata string, block int) {
+	var count uint64 = 5000000
+	var entries Entries = make([]byte, count*35)
+	bn := make([]byte, 3)
+	var lookups []LookupFile
+	var iterations uint64
+	var blockNum = startBlock
+	var interrupt bool
+	filename := fmt.Sprintf(".lookups_%d.tmp", len(lookups))
+	fileTmp, _ := os.OpenFile(filename, os.O_APPEND|os.O_CREATE|os.O_RDWR, 0644)
+	for !interrupt {
+		blockHash := rawdb.ReadCanonicalHash(db, blockNum)
+		body := rawdb.ReadBody(db, blockHash, blockNum)
+
+		if body == nil {
+			log.Info("Now Inserting to file")
+			insertInFileForLookups2(fileTmp, entries, iterations)
+			lookups = append(lookups, LookupFile{nil, nil, 35})
+			iterations = 0
+			entries = nil
+			fileTmp.Close()
+			break
+		}
+
+		select {
+		case interrupt = <-interruptCh:
+			log.Info("interrupted, please wait for cleanup...")
+		default:
+		}
+		bn[0] = byte(blockNum >> 16)
+		bn[1] = byte(blockNum >> 8)
+		bn[2] = byte(blockNum)
+
+		for _, tx := range body.Transactions {
+			copy(entries[35*iterations:], tx.Hash().Bytes())
+			copy(entries[35*iterations+32:], bn)
+			iterations++
+			if iterations == count {
+				log.Info("Now Inserting to file")
+				insertInFileForLookups2(fileTmp, entries, iterations)
+				entries = nil
+				lookups = append(lookups, LookupFile{nil, nil, 35})
+				iterations = 0
+				fileTmp.Close()
+				filename = fmt.Sprintf(".lookups_%d.tmp", len(lookups))
+				fileTmp, _ = os.OpenFile(filename, os.O_APPEND|os.O_CREATE|os.O_RDWR, 0644)
+			}
+		}
+		blockNum++
+		if blockNum%100000 == 0 {
+			log.Info("Processed", "blocks", blockNum, "iterations", iterations)
+			var m runtime.MemStats
+			runtime.ReadMemStats(&m)
+			log.Info("Memory", "alloc", int(m.Alloc/1024), "sys", int(m.Sys/1024), "numGC", int(m.NumGC))
+		}
+	}
+	batch := db.NewBatch()
+	h := &Heap{}
+	heap.Init(h)
+	for i := range lookups {
+		file, err := os.Open(fmt.Sprintf(".lookups_%d.tmp", i))
+		check(err)
+		lookups[i].file = bufio.NewReader(file)
+		lookups[i].buffer = make([]byte, bufferLen)
+		check(err)
+		n, err := lookups[i].file.Read(lookups[i].buffer)
+		if n != bufferLen {
+			lookups[i].buffer = lookups[i].buffer[:n]
+		}
+		heap.Push(h, HeapElem{lookups[i].buffer[:35], i})
+	}
+
+	for !interrupt && len(*h) != 0 {
+		val := (heap.Pop(h)).(HeapElem)
+		if lookups[val.index].pos == uint64(bufferLen) {
+			if val.val[32] != 0 {
+				err := batch.Put(dbutils.TxLookupPrefix, val.val[:32], common.CopyBytes(val.val[32:]))
+				check(err)
+			} else {
+				err := batch.Put(dbutils.TxLookupPrefix, val.val[:32], common.CopyBytes(val.val[33:]))
+				check(err)
+			}
+			n, _ := lookups[val.index].file.Read(lookups[val.index].buffer)
+			iterations++
+			if n == 0 {
+				os.Remove(fmt.Sprintf(".lookups_%d.tmp", val.index))
+			} else {
+				if n != bufferLen {
+					lookups[val.index].buffer = lookups[val.index].buffer[:n]
+				}
+				lookups[val.index].pos = 35
+				heap.Push(h, HeapElem{lookups[val.index].buffer[:35], val.index})
+			}
+			continue
+		} else {
```

Comment:
You can remove this `else` branch, because corresponding `if` always does `continue`